src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -307,10 +313,24 @@ public static LSP.Range TextSpanToRange(TextSpan textSpan, SourceText text)
                 Range = MappedSpanResultToRange(mappedSpan)
             };
 
-            static async Task<LSP.Location> ConvertTextSpanToLocation(Document document, TextSpan span, CancellationToken cancellationToken)
+            static async Task<LSP.Location> ConvertTextSpanToLocation(
+                Document document,
+                TextSpan span,
+                bool isStale,
+                CancellationToken cancellationToken)
             {
                 var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
 
+                if (isStale)
+                {
+                    // in the case of a stale item, the span may be out of bounds of the document. Cap
+                    // us to the end of the document as that's where we're going to navigate the user
+                    // to.
+                    span = TextSpan.FromBounds(
+                        Math.Min(text.Length, span.Start),
+                        Math.Min(text.Length, span.End));
```

Comment:
stale results are possible as of: https://github.com/dotnet/roslyn/pull/52380

Reply:
Do we have the same clamping happening in the non-LSP implementation as well? Is there way to unify that?

Reply:
yes, we do.  i'll see if i can share the code somehow.